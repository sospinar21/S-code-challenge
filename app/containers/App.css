.App {
}

.SwaggerDebug {
  margin: 10px;
  max-height: 400px;
  overflow: auto;
}

.todos {
  width: 100%;
  height: 100vh;
  background-color: red;
  padding: 50px;
}

.todo-div{
  width: 100%;
  height: 100vh;
  background-color: red;
}
