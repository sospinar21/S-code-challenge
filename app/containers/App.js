import React from 'react';
import { connect } from 'react-redux';
import style from './App.css';

class App extends React.Component {
  constructor() {
    super()

    this.state = {
      todos:[]
    };
  }

  async componentDidMount() {
    const todos = await this.getTodos();
  };

  getTodos = async () => {
    const response = await fetch('https://api.awc.dance/codechallenge')
    const data = await response.json();
    await this.addTodosToState(data);
    return await data;
  };

  addTodosToState = (data) => {
    const todos = data.map(todo => todo);
    this.setState({todos});
  }

  displayTodos = () => {
    const todos = this.state.todos.map((todo, index) => {
      const completed = todo.completed !== null ? style.tododiv : style.completed;
      const completedText = todo.completed !== null ? style.completedText : style.noCompleted;
      return (
      <div>
        <div key={'todo' + index} className={completed}>
          <h2 className={completedText}>Name: {todo.name}</h2>
          <p className={style.id}>Project's Id: {todo.id}</p>
          <a href={style.url} className={style.url}>Link {todo.url}</a>
        </div>
        <hr/>
      </div>
      )
    }) 
    return todos;
  }

  render() {
    const { swagger } = this.props;
    const todos = this.displayTodos();
    return (
      <div className={style.App}>
        <div className={style.header}>
          <h1 className={style.mainTitle}>ToDos</h1>
        </div>
        <div className={style.todos}>
          {todos}
        </div>
        <pre className={style.SwaggerDebug}>
        </pre>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return { swagger: state.swagger };
}

export default connect(mapStateToProps)(App);
