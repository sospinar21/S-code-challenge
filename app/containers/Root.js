import React from 'react';
import { Provider } from 'react-redux';
import App from './App';

export default class Root extends React.Component {
  render() {
    const { store } = this.props;
    return (
      <Provider store={store}>
        <App />
      </Provider>
    );
  }
}
