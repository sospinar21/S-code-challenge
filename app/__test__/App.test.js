import {App, mapStateToProps} from '../containers/App'
import React from 'react';
import ReactDOM from 'react-dom';
import { mount, shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


describe('community', () => {
  let app;

  beforeEach(() => {

    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      status: 200,
      json: () => Promise.resolve({data: {}})
    }));
    app = mount(<App />);

  });

  it('matches snapshot', () => {
    expect(app).toMatchSnapshot();
  });

  describe('addTodosToState', () => {

    it('updates the state with the todos', async () => {
      app = mount(<App />);
      await app.instance().addTodosToState(['todo1', 'todo2']);
      expect(app.state().todos).toEqual(['todo1', 'todo2'])
    });
  });

});