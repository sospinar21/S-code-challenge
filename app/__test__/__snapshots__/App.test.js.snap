// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`community matches snapshot 1`] = `undefined`;
